test/contrib/bigquery_gcloud_test.py

```diff
@@ -218,3 +223,68 @@ def test_run_query(self):
         task.run()
 
         self.assertTrue(self.bq_client.table_exists(self.table))
+
+
+@attr('gcloud')
+class BigQueryLoadAvroTest(unittest.TestCase):
+    def _produce_test_input(self):
+        schema = avro.schema.parse("""
+        {
+            "name": "TestQueryTask_record",
+            "type": "record",
+            "doc": "The description",
+            "fields": [
+                {"name": "col0", "type": "int", "doc": "The bold"},
+                {"name": "col1", "type": {
+                    "name": "inner_record",
+                    "type": "record",
+                    "doc": "This field shall be an inner",
+                    "fields": [
+                        {"name": "inner", "type": "int", "doc": "A inner field"}
+                    ]
+                }, "doc": "This field shall be an inner"},
+                {"name": "col2", "type": "int", "doc": "The beautiful"}
+            ]
+        }""")
+        writer = DataFileWriter(open("tmp.avro", "wb"), DatumWriter(), schema)
+        writer.append({'col0': 1000, 'col1': {'inner': 1234}, 'col2': 1001})
+        writer.close()
+        self.gcs_client.put("tmp.avro", self.gcs_dir_url)
+        # FIXME don't need a tmp.avro; use GCS API like objects().insert that accepts file contents in a string:
+        # https://developers.google.com/resources/api-libraries/documentation/storage/v1/python/latest/storage_v1.objects.html#insert
+
+    def setUp(self):
+        self.gcs_client = gcs.GCSClient(CREDENTIALS)
+        self.bq_client = bigquery.BigQueryClient(CREDENTIALS)
+
+        self.table_id = "avro_bq_table"
+        self.gcs_dir_url = 'gs://' + BUCKET_NAME + "/foo/tmp.avro"
+        self._produce_test_input()
+        self.addCleanup(self.bq_client.delete_dataset, bigquery.BQDataset(PROJECT_ID, DATASET_ID, EU_LOCATION))
+        self.addCleanup(self.gcs_client.remove, self.gcs_dir_url)
```

Comment:
This should come before producing test input or immediately after it, otherwise there's code in between that can crash and thus prevent adding of the cleanup.